sdg/schemas/SchemaInputBase.py

```diff
@@ -170,3 +171,27 @@ def get_value_translation(self, field):
             raise ValueError(field + " field does not have options element")
 
         return {x['enum'][0]: x['translation_key'] for x in options}
+
+
+    def add_field_order(self, field):
+        """Add a field to the list, in case an output needs a field order.
+
+        Parameters
+        ----------
+        field : string
+            The name of a field to add to the list
+        """
+        self.field_order.append(field)
+
+
+    def get_field_order(self):
+        """Get the list of fields in the preserved order.
+
+        Returns
+        -------
+        list
+            A list of field names in a particular order
+        """
+        if len(self.field_order) == 0:
+            raise Exception('The schema input did not preserve the order of fields.')
```

Comment:
Maybe the function should default to something instead of throwing an exception?

Reply:
Good point, it can just default to the keys. They'll be in a random order, but I guess that's OK.